diff --git a/linux-image.sh b/linux-image.sh
index 1fd03e9..b1aef33 100755
--- a/linux-image.sh
+++ b/linux-image.sh
@@ -1,12 +1,14 @@
 #!/usr/bin/env bash
 
+# This file is originally from https://github.com/rust-embedded/cross but modified to work with Ubuntu 20.10, gtk and libhandy.
+
 set -x
 set -euo pipefail
 
 main() {
     # arch in the rust target
     local arch="${1}" \
-          kversion=4.19.0-14
+          kversion=5.8.0-25
 
     local debsource="deb http://http.debian.net/debian/ buster main"
     debsource="${debsource}\ndeb http://security.debian.org/ buster/updates main"
@@ -21,7 +23,7 @@ main() {
     case "${arch}" in
         aarch64)
             arch=arm64
-            kernel="${kversion}-arm64"
+            kernel="${kversion}-generic"
             ;;
         armv7)
             arch=armhf
@@ -84,7 +86,7 @@ main() {
             ;;
         x86_64)
             arch=amd64
-            kernel="${kversion}-amd64"
+            kernel="${kversion}-generic"
             ;;
         *)
             echo "Invalid arch: ${arch}"
@@ -110,8 +112,8 @@ main() {
     done
 
     # Download packages
-    mv /etc/apt/sources.list /etc/apt/sources.list.bak
-    echo -e "${debsource}" > /etc/apt/sources.list
+    cp /etc/apt/sources.list /etc/apt/sources.list.bak
+    echo -e "${debsource}" >> /etc/apt/sources.list
 
     # Old ubuntu does not support --add-architecture, so we directly change multiarch file
     if [ -f /etc/dpkg/dpkg.cfg.d/multiarch ]; then
